os/arch/arm/src/amebad/amebad_serial.c

```diff
@@ -571,6 +573,22 @@ static int rtl8721d_up_ioctl(FAR struct uart_dev_s *dev, int cmd, unsigned long
 		serial_control_loopback(sdrv[uart_index_get(priv->tx)], *(bool *)arg);
 		break;
 
+	case TIOCS_AVAIL:
+		g_tc_rxavailable = 1;
+		g_tc_rxavailable_res = 0;
+		rtl8721d_up_send(dev, 'a');
+		sleep(1);
+		ret = g_tc_rxavailable_res;
+		break;
+
+	case TIOCS_READY:
+		ret = rtl8721d_up_txready(dev);
+		break;
+
+	case TIOCS_EMPTY:
+		ret = rtl8721d_up_txempty(dev);
+		break;
```

Comment:
1. How about other boards like imxrt? Does it return invalid or not supported or already impleneted?
2. Are they only used in the tc for coverage? Then let's cover them with TC config.

Reply:
fill add this in other boards after finalizing for this board